scylla/tests/ccm_integration/keep_alive.rs

```diff
@@ -0,0 +1,276 @@
+use crate::common::utils::test_with_3_node_cluster;
+use scylla::client::session::Session;
+use scylla::client::session_builder::SessionBuilder;
+
+use scylla_proxy::{
+    Condition, ProxyError, Reaction, RequestOpcode, RequestReaction, RequestRule, ShardAwareness,
+    WorkerError,
+};
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::sync::mpsc::{self, UnboundedReceiver};
+use tokio::time::sleep;
+
+/// Waits for a specified number of keep-alive messages to be received within a given timeout period.
+///
+/// # Arguments
+///
+/// * `request_rx` - An `UnboundedReceiver` that receives the keep-alive messages.
+/// * `expected_number_of_keep_alive` - The number of keep-alive messages expected to be received.
+/// * `timeout` - The maximum duration to wait for the expected number of keep-alive messages.
+///
+/// # Panics
+///
+/// This function will panic if the number of received keep-alive messages does not match the
+/// expected number within the timeout period.
+///
+async fn wait_for_keep_alive<T>(
+    mut request_rx: UnboundedReceiver<T>,
+    expected_number_of_keep_alive: u32,
+    timeout: Duration,
+) {
+    let mut total_keep_alives: u32 = 0;
+    let start = tokio::time::Instant::now();
+    while total_keep_alives < expected_number_of_keep_alive && start.elapsed() < timeout {
+        if let Ok(_t) = request_rx.try_recv() {
+            total_keep_alives += 1;
+        }
+        sleep(Duration::from_millis(1)).await;
+    }
+    println!("Total keep alive: {}", total_keep_alives);
+    assert_eq!(total_keep_alives, expected_number_of_keep_alive);
+}
+
+#[tokio::test]
+#[cfg(not(scylla_cloud_tests))]
+async fn node_should_go_down_gracefully_when_connection_closed_during_heartbeat() {
+    let res = test_with_3_node_cluster(
+        ShardAwareness::QueryNode,
+        |proxy_uris, translation_map, mut running_proxy| async move {
+            let _session: Session = SessionBuilder::new()
+                .known_node(proxy_uris[0].as_str())
+                .address_translator(Arc::new(translation_map))
+                .keepalive_interval(Duration::from_millis(20))
+                .build()
+                .await
+                .unwrap();
+
+            // TODO: No way to get node status, no as in java-driver
+
```

Comment:
WDYM by node status? Do you need information if the driver is connected to a given node?
If so you can get it with `session.cluster_state().get_nodes_info()[node_idx].is_connected()`